File: binarySearch.cpp
#include<bits/stdc++.h>
using namespace std;

int binarySearch(int arr[],int l,int r,int x){
    while(l <= r){
        int m = (l + (r-1))/2;
        if (arr[m] == x)
        {
            return m+1;
        }
        if (arr[m] < x)
        {
            l = m + 1;
        }
        else
        {
            r = m - 1;
        }
        
    }
    return -1;
}

//driver code 
int main(){
    //intialize
    int a[] = {8,65,4,3,3,9,7,1};
    int x = 3;
    int n = sizeof(a)/sizeof(a[0]);

    int result = binarySearch(a,0,n-1,x);
    (result == -1)
        ? cout << "NOT PRESENT IN THE ARRAY "
        : cout << "PRESNT AT POSITION " << result;
    return 0;
}

File: binarySearch.java
public class binarySearch {
    static int binSearch(int arr[], int l, int r, int x){
        while(l <= r){
            int mid = (l + r)/2;
            if(arr[mid] == x)
                return mid;
            else if(arr[mid] > x)
                r = mid-1;
            else
                l = mid+1;
        }
        return -1;
    }
    public static void main(String[] args) {
        int a[] = {34,45,56,67,78,89};
        int n = a.length;
        int x = 67;
        int result = binSearch(a,0,(n-1),x);
        if(result == -1)
            System.out.println("NUMBER IS NOT PRESENT IN LIST");
        else
            System.out.println("the number "+x+" is present at location "+(result+1));
    }
}
